sim/dex.ts

```diff
@@ -653,11 +659,13 @@ class ModdedDex {
 		let statName: keyof StatsTable;
 		for (statName in modStats) {
 			const stat = baseStats[statName];
-			modStats[statName] = Math.floor(Math.floor(2 * stat + set.ivs[statName] + Math.floor(set.evs[statName] / 4)) * set.level / 100 + 5);
+			modStats[statName] = Math.floor(Math.floor(2 * stat + set.ivs[statName] +
```

Comment:
```suggestion
			const tr = this.trunc;
			modStats[statName] = tr(tr(2 * stat + set.ivs[statName] + tr(set.evs[statName] / 4)) * set.level / 100 + 5);
```

You'll need to also pull `trunc` out of `battle.ts` and into `dex.ts`.

Reply:
(n.b. this might require some changes to how `trunc` is implemented; I remember there's some magic with how it works in custom games)

Reply:
Done. It seems we use `Math.floor` in a lot of places we should be using `trunc`... (out of scope for this PR, but maybe worth fixing?)